exchanges/bitfinex/bitfinex.go

```diff
@@ -118,6 +119,140 @@ func (b *Bitfinex) GetPlatformStatus() (int, error) {
+		tempData, ok := data[x].([]interface{})
+		if !ok {
+			return nil, fmt.Errorf("%w for all sym", errTypeAssert)
+		}
+		var check bool
+		tempResp.Symbol, check = tempData[1].(string)
+		if !check {
+			return nil, fmt.Errorf("%w for symbol data", errTypeAssert)
+		}
+		tempFloatData, check := tempData[2].([]interface{})
+		if !check {
+			return nil, fmt.Errorf("%w for symbol data", errTypeAssert)
+		}
+		if len(tempFloatData) < 4 {
+			return nil, errors.New("invalid data received")
+		}
+		tempResp.TradableBalance, ok = tempFloatData[0].(float64)
+		if !ok {
+			return nil, fmt.Errorf("%w for TradableBalance", errTypeAssert)
+		}
+		tempResp.GrossBalance, ok = tempFloatData[1].(float64)
+		if !ok {
+			return nil, fmt.Errorf("%w for GrossBalance", errTypeAssert)
+		}
+		tempResp.BestAskAmount, ok = tempFloatData[2].(float64)
+		if !ok {
+			return nil, fmt.Errorf("%w for BestAskAmount", errTypeAssert)
+		}
+		tempResp.BestBidAmount, ok = tempFloatData[3].(float64)
+		if !ok {
+			return nil, fmt.Errorf("%w for BestBidAmount", errTypeAssert)
+		}
+		resp = append(resp, tempResp)
+	}
+	return resp, nil
+}
+
+func defaultMarginV2Info(data []interface{}) (MarginInfoV2, error) {
+	var resp MarginInfoV2
+	var ok bool
+	resp.Symbol, ok = data[1].(string)
+	if !ok {
+		return resp, fmt.Errorf("%w for symbol", errTypeAssert)
+	}
+	tempData, check := data[2].([]interface{})
+	if !check {
+		return resp, fmt.Errorf("%w for symbol data", errTypeAssert)
+	}
+	if len(tempData) < 4 {
+		return resp, errors.New("invalid data received")
+	}
+	resp.TradableBalance, ok = tempData[0].(float64)
+	if !ok {
+		return resp, fmt.Errorf("%w for TradableBalance", errTypeAssert)
+	}
+	resp.GrossBalance, ok = tempData[1].(float64)
+	if !ok {
+		return resp, fmt.Errorf("%w for GrossBalance", errTypeAssert)
+	}
+	resp.BestAskAmount, ok = tempData[2].(float64)
+	if !ok {
+		return resp, fmt.Errorf("%w for BestAskAmount", errTypeAssert)
+	}
+	resp.BestBidAmount, ok = tempData[3].(float64)
+	if !ok {
+		return resp, fmt.Errorf("%w for BestBidAmount", errTypeAssert)
+	}
+	return resp, nil
+}
+
+// GetV2MarginInfo gets v2 margin info for a symbol provided
+// symbol: base, sym_all, any other trading symbol example tBTCUSD
+func (b *Bitfinex) GetV2MarginInfo(symbol string) ([]MarginInfoV2, error) {
+	var data []interface{}
+	err := b.SendAuthenticatedHTTPRequestV2(exchange.RestSpot, http.MethodPost,
+		bitfinexV2MarginInfo+symbol,
+		nil,
+		&data,
+		getMarginInfoRate)
+	if err != nil {
+		return nil, err
+	}
+	var tempResp MarginInfoV2
+	switch symbol {
+	case "base":
+		tempResp, err = baseMarginInfo(data)
+		if err != nil {
+			return nil, fmt.Errorf("%v - %s: %w", b.Name, symbol, err)
+		}
+	case "sym_all":
+		resp, err := symbolMarginInfo(data)
+		return resp, err
+	default:
+		tempResp, err = defaultMarginV2Info(data)
+		if err != nil {
+			return nil, fmt.Errorf("%v - %s: %w", b.Name, symbol, err)
+		}
+	}
+	return []MarginInfoV2{tempResp}, nil
```

Comment:
This has the same issue as the previous comment https://github.com/thrasher-corp/gocryptotrader/pull/681#discussion_r659397327
Also the shadow declaration of err in `sym_all`